Change code block to handlebars

amyrlam opened the pull request:
## What it does
<!--- Tell us what this fix does in a few sentences. E.g. "This updates the header title's font color to Ember Orange." -->

ember-prism addon can only take triple curly code blocks in this format:

```
```handlebars
{{something-cool}}
<close-triple-ticks>
```

if you use the shorthand `hbs`, it won't work, there will be a build error in your terminal

```
```hbs
{{something-cool}}
<close-triple-ticks>
```

(writing it with `<close-triple-ticks>` so that it will 

## Related Issue(s)
<!--- Please provide the issue(s) to which this pull request relates to or which issue it closes. E.g. "Closes #1234" -->

Resolves: https://github.com/ember-learn/ember-blog/issues/1005

amyrlam committed:
Change html code block to handlebars

The code block must be written as `handlebars`, `hbs` will not work.

Resolves: https://github.com/ember-learn/ember-blog/issues/1005 (more context there too)

amyrlam committed:
Add note about code blocks

NullVoxPopuli commented:
You can use hbs and better highlighting with this: (I did the name fix earlier today): https://github.com/NullVoxPopuli/website/pull/45/files#diff-322f0482bd28fc40f84fcb0aaf208fef7fb63082d462cf02d11f46e59c531c24R6

amyrlam committed:
Set up ember-prism for hbs in code blocks

Also, upgrade dependencies re:

```
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
```

amyrlam commented:
@NullVoxPopuli thank you!

- This works but I'm not sure if ember-prism should be installed here or somewhere up the chain in empress. But didn't see anything for "prism" in https://github.com/empress/empress-blog-casper-template. I could be missing something, do you know, @mansona or others?
- I can separate out the `npx browserslist@latest --update-db` change, in debugging was doing a few things.

NullVoxPopuli commented:
I mean, I think this should be a default in ember-prism, but I don't remember why it's not 🤔

This is the PR that introduced it: https://github.com/shipshapecode/ember-prism/pull/279

But I don't see a discussion about it. 🤔 

@rwwagner90 thoughts on making prism-glimmer default? I'm not sure how it would work - not supes familiar with how prism is bundled / is running

ijlee2 reviewed (changes requested):
@amyrlam I think you added the `ember-prism` package because, without it, running `start()` will result in the error,

```
Error: Could not find module `ember-prism` imported from `ember-blog/app`
```

However, `ember-prism` already gets pulled in through `empress-blog-ember-template`. I'm unsure what having a separate package will mean for us in the long run.

For simplicity, can we:

- Revert the changes to `app/app.js`, `ember-cli-build.js`, `package.json`, and `package-lock.json`
- Use `handlebars` instead of `html` in the file `content/the-ember-times-issue-184.md`

ijlee2 commented on CONTRIBUTING.md:
````diff
@@ -159,6 +159,12 @@ Thank you for contributing to The Ember Times! Let us know if you can be a guest
+1. Note: If you are making a Handlebars code block, you must use `handlebars` instead of `hbs`. For example:
+
+```handlebars
+{{something-cool}}
````
I think the following code example may illustrate when to use `handlebars` better?

```suggestion
{{!-- app/templates/application.hbs --}}

<h1>Hello world!</h1>

{{outlet}}
```

amyrlam commented:
@ijlee2 Makes sense... was thinking to update upstream. It looks like upstream is using `ember-showdown-prism` though, which builds on `ember-prism` https://github.com/ember-learn/empress-blog-ember-template/blob/master/package.json#L43 ... let me do some investigation there. Thanks for the feedback!

amyrlam committed:
Update CONTRIBUTING.md

Co-authored-by: Isaac Lee <16869656+ijlee2@users.noreply.github.com>

NullVoxPopuli commented:
Fwiw, embroider requires packages that are imported in app code to be declared in package.json

jenweber commented on content/the-ember-times-issue-184.md:
````diff
@@ -41,7 +41,7 @@ The blog post argues against the `disabled` attribute in `<button>` in favor of
 This modifier implements accessible and usable button-disabling by adding `aria-disabled="true"` to the target element when the `when` named argument is truthy. It will also stop any clicks on the button from propagating and ensures form submission does not trigger.
 
-```html
+```hbs
````
This should be `handlebars` instead. Maybe it will work after that change?

In the learning team discussion - if there is an issue with prism, bugfixes would need to go upstream in https://github.com/empress/ember-showdown-prism

NullVoxPopuli replied:
if hifi-glimmer is enabled, `hbs` works fine

mansona commented on app/app.js:
```diff
@@ -2,7 +2,9 @@ import Application from '@ember/application';
 import Resolver from 'ember-resolver';
 import loadInitializers from 'ember-load-initializers';
 import config from 'ember-blog/config/environment';
+import { setup } from 'ember-prism';
```
just to follow up to what @jenweber said, we should not be doing anything ember-prism directly in ember-blog. This is all handled by the template and the addon that Jen already mentioned 👍

amyrlam commented:
@jenweber @mansona i'm not sure exactly what you are suggesting, can u clarify what u mean?

suggestion: why don't i go back to @ijlee2's suggestion https://github.com/ember-learn/ember-blog/pull/1010#pullrequestreview-720333259 and file an issue on ember-showdown-prism, which doesn't seem to be bringing in the latest changes to ember-prism yet that @NullVoxPopuli did. 

i'm going to work on the times first and get back to this task next. sorry for the delay have been sick.

amyrlam commented:
i've made this issue upstream, to hopefully look at later: https://github.com/empress/ember-showdown-prism/issues/17

i tagged the four of you on https://github.com/ember-learn/ember-blog/pull/1019 and am closing this PR in favor of that one!
